File: 724. Find Pivot Index/Solution.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var pivotIndex = function(nums) {
    let sumLeft = 0;
    let sumRight = nums.reduce((a, b) => a+b);
    for(let i = 0; i < nums.length; i++) {
        sumRight -= nums[i];
        if(sumLeft === sumRight) {
            return i;
        }
        sumLeft += nums[i];
    }
    return -1;
};

File: 724. Find Pivot Index/Solution.cpp
class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        int sumLeft = 0;
        int sumRight = std::accumulate(nums.begin(), nums.end(), 0);
        for(int i = 0; i < nums.size(); i++) {
            sumRight -= nums[i];
            if(sumLeft == sumRight) {
                return i;
            }
            sumLeft += nums[i];
        }
        return -1;
    }
};
